INT-4503: Fix assertion in HttpMessageHandlerSpec

imitbn opened the pull request:
JIRA: https://jira.spring.io/browse/INT-4503

Custom ErrorHandler can't be used if external RestTemplate is provided

* Replace isClientSet() with !isClientSet() for isTrue assertion in errorHandler method

artembilan commented on spring-integration-http/src/main/java/org/springframework/integration/http/dsl/HttpMessageHandlerSpec.java:
```diff
@@ -75,8 +75,7 @@ public HttpMessageHandlerSpec requestFactory(ClientHttpRequestFactory requestFac
 	public HttpMessageHandlerSpec errorHandler(ResponseErrorHandler errorHandler) {
-		Assert.isTrue(this.isClientSet(),
-				"the 'errorHandler' must be specified on the provided 'restTemplate'");
+		Assert.isTrue(!isClientSet(), "the 'errorHandler' must be specified on the provided 'restTemplate'");
```
Good. Thank you!

Would you mind to modify `HttpDslTests` to cover this change?

Also consider, please, fix `requestFactory()` condition above to use this `isClientSet()` as well.

And add you name to the `@author` list of this class and test one, if you really are going to modify that as well.

imitbn committed:
INT-4503: Fix assertion in HttpMessageHandlerSpec

JIRA: https://jira.spring.io/browse/INT-4503

Custom ErrorHandler can't be used if external RestTemplate is provided

* Replace isClientSet() with !isClientSet() for isTrue assertion in errorHandler method
* Replace assertion of restTemplate isNull with assertion !isClientSet() isTrue in requestFactory method
* Add test coverage of errorHandler in HttpDslTests

artembilan reviewed (approved):
@imitbn ,

you change has slipped between our fingers because you squashed commits.
Would be great to just add new commits and we will got a notification for new changes.

So far your fix looks good and we are merging it...

artembilan commented:
Merged as https://github.com/spring-projects/spring-integration/commit/865e2162e520f8be6b26f81472a1ced22e8e1c0f and cherry-picked to `5.0.x`

Thank you for the contribution.
Looking forward for more!
